docs/debugging.md

```diff
@@ -0,0 +1,90 @@
+# Debugging Gate Failures
+
+## Blue Ocean
+[Blue ocean](https://rpc.jenkins.cit.rackspace.net/blue/organizations/jenkins/pipelines) is the new UI for [Jenkins](https://rpc.jenkins.cit.rackspace.net/), it makes it easier to find the point where
+a job failed, without trawling through the console. To use blue ocean, follow
+the link from a github pull request status to a job, then click open "Open Blue
+Ocean" from the left sidebar.
+
+<img src="images/debugging/sidebar.png" width=200>
+
+Blue ocean displays a horizontal pipeline of stages. Stages that failed will
+be shown in red. Clicking a stage reveals a list of steps beneath,
+steps that failed will also be marked as red. Expanding a step will show it's logs.
+
+![](images/debugging/blueocean.png)
+
+Using this method is much quicker for finding the failure point than reading the
+console.
+
+## The Console
+Jenkins collects all the output of commands it executes, these outputs are
+streamed to the console for each Job. This includes logging for Jenkins
```

Comment:
...*Jenkins collects all of the output from commands it executes, and these outputs*...